File: in_C++/1896A_jagged_swaps.cpp
#include <bits/stdc++.h>
#include <algorithm>
using namespace std;

int main()
{
    int t;
    cin >> t;

    while (t--)
    {
        int num;
        cin >> num;
        int temp;

        int arr[num];
        for (int i = 0; i < num; i++)
        {
            cin >> arr[i];
        }
        for (int i = 0; i < num; i++)
        {
            for (int j = 2; j < num - i; j++)
            {
                if (arr[j - 1] > arr[j - 2] and arr[j - 1] > arr[j])
                {
                    temp = arr[j - 1];
                    arr[j - 1] = arr[j];
                    arr[j] = temp;
                }
            }
        }

        if (is_sorted(arr, arr + num))
        {
            cout << "YES" << endl;
        }
        else
        {
            cout << "NO" << endl;
        }
    }
    return 0;
}

File: in_Python/1896A_jagged_swaps.py
t = int(input())
for i in range(t):
    num = int(input())
    array = list(map(int,input().split()))
    for i in range(num):
        for j in range(2,num-i):
            if (array[j-1] > array[j] and array[j-1] > array[j-2]):
                temp = array[j-1]
                array[j-1] = array[j]
                array[j] = temp
    if array == sorted(array):
        print("YES")
    else:
        print("NO")